pywr/solvers/cython_glpk.pyx

```diff
@@ -577,15 +518,8 @@ cdef class CythonGLPKSolver:
 
         t0 = time.perf_counter()
 
-        # Apply presolve if required
-        if self.use_presolve and not self.has_presolved:
-            self.smcp.presolve = GLP_ON
-            self.has_presolved = True
-        else:
-            self.smcp.presolve = GLP_OFF
-
         # Set the basis for this scenario
-        self._set_basis(scenario_index.global_id)
+        self.basis_manager.set_basis(self.prob, self.is_first_solve, scenario_index.global_id)
```

Comment:
Can we abstract the whole section on solving the model and error handling (522-549 here) into the base class?